### PR TITLE
PD-859: Adds darkMode prop to IconButton

### PR DESCRIPTION
**Pull request by brookescarlett**
## ✍️ Proposed changes

- Adds `darkMode` to IconButton
- Refactors IconButton to use Box
- Updates Copy component to pass `darkMode` prop to IconButton 

🎟 _Jira ticket:_ [PD-859](https://jira.mongodb.org/browse/PD-859)

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README

### CODE REVIEW

**Comment by brookescarlett** [comment]
@lazytype Where do you think is an appropriate place to document the fact that TS raises an error about the lack of "as" prop supplied, simply because it's the last overload, when really the error is a call to fixing the aria-label/aria-labelledby sitaution?

**Comment by lazytype** [comment]
> @lazytype Where do you think is an appropriate place to document the fact that TS raises an error about the lack of "as" prop supplied, simply because it's the last overload, when really the error is a call to fixing the aria-label/aria-labelledby sitaution?

I think we could add aria-label/aria-labelledby explicitly as props in the README. The error message itself says that no overloads match and I think at some point or another every TypeScript developer needs to learn that this means the error doesn't necessarily correspond to the last overload.

**Review by lazytype — CHANGES_REQUESTED** [review]
adding back to your queue for reply

**Comment by brookescarlett** [comment]
We do explicitly mention this in the readme already: https://github.com/mongodb/leafygreen-ui/pull/603/files#diff-4076331f3aaf45bd74544aee10890545R65-R69. Is this sufficient, or should we add more coverage?

**Comment by lazytype** [comment]
> We do explicitly mention this in the readme already: https://github.com/mongodb/leafygreen-ui/pull/603/files#diff-4076331f3aaf45bd74544aee10890545R65-R69. Is this sufficient, or should we add more coverage?

oh, didn't see it in the diff there. might help to have the prop be a row in the table to that points to the section below

File: packages/icon-button/README.md [hunk 1]
```diff
@@ -51,16 +51,17 @@ import IconButton from '@leafygreen-ui/icon-button';
...
+| `href`                            | `string`                     | If a href is supplied, the component renders inside of an `a` tag instead of inside of a `button` tag.            |           |
+| `className`                       | `string`                     | Adds a className to the class attribute on the container element.                                                 |           |
+| `children`                        | `node`                       | Content rendered inside of the `<IconButton />` component                                                         |           |
+| `size`                            | `default`, `large`, `xlarge` | Determines the size of the IconButton                                                                             | `default` |
+| `active`                          | `boolean`                    | Determines whether the `<IconButton />` will appear `active`                                                      | `false`   |
+| `aria-label` or `aria-labelledby` | `string`                     | One of these must be provided. See note below                                                                     |           |
```
Comment (brookescarlett) [inline packages/icon-button/README.md:64]: Does this work @lazytype

**Review by lazytype — APPROVED** [review]

File: packages/icon-button/src/IconButton.tsx [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
+const Mode = {
   Light: 'light',
   Dark: 'dark',
 } as const;
 
-type Variant = typeof Variant[keyof typeof Variant];
+type Mode = typeof Mode[keyof typeof Mode];
 
-export { Variant };
+export { Mode };
```
Comment (lazytype) [inline packages/icon-button/src/IconButton.tsx:16]: don't think we need this export

**Pull request merged**